Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -243,6 +249,197 @@ extension Processor {
+    switch kind {
+    case .startOfSubject: return currentPosition == subjectBounds.lowerBound
+
+    case .endOfSubjectBeforeNewline:
+      if currentPosition == subjectBounds.upperBound { return true }
+      switch semanticLevel {
+      case .graphemeCluster:
+        return input.index(after: currentPosition) == subjectBounds.upperBound
+         && input[currentPosition].isNewline
+      case .unicodeScalar:
+        return input.unicodeScalars.index(after: currentPosition) == subjectBounds.upperBound
+         && input.unicodeScalars[currentPosition].isNewline
+      }
+
+    case .endOfSubject: return currentPosition == subjectBounds.upperBound
+
+    case .resetStartOfMatch:
+      // FIXME: Figure out how to communicate this out
+      throw Unsupported(#"\K (reset/keep assertion)"#)
+
+    case .firstMatchingPositionInSubject:
+      // TODO: We can probably build a nice model with API here
+
+      // FIXME: This needs to be based on `searchBounds`,
+      // not the `subjectBounds` given as an argument here
+      // (Note: the above fixme was in reference to the old assert function API.
+      //   Now that we're in processor, we have access to searchBounds)
+      return false
+
+    case .textSegment: return input.isOnGraphemeClusterBoundary(currentPosition)
+
+    case .notTextSegment: return !input.isOnGraphemeClusterBoundary(currentPosition)
+
+    case .startOfLine:
+      // FIXME: Anchor.startOfLine must always use this first branch
+      // The behavior of `^` should depend on `anchorsMatchNewlines`, but
+      // the DSL-based `.startOfLine` anchor should always match the start
+      // of a line. Right now we don't distinguish between those anchors.
+      if anchorsMatchNewlines {
+        if currentPosition == subjectBounds.lowerBound { return true }
+        switch semanticLevel {
+        case .graphemeCluster:
+          return input[input.index(before: currentPosition)].isNewline
+        case .unicodeScalar:
+          return input.unicodeScalars[input.unicodeScalars.index(before: currentPosition)].isNewline
+        }
+      } else {
+        return currentPosition == subjectBounds.lowerBound
+      }
+  
+      case .endOfLine:
+        // FIXME: Anchor.endOfLine must always use this first branch
+        // The behavior of `$` should depend on `anchorsMatchNewlines`, but
+        // the DSL-based `.endOfLine` anchor should always match the end
+        // of a line. Right now we don't distinguish between those anchors.
+        if anchorsMatchNewlines {
+          if currentPosition == subjectBounds.upperBound { return true }
+          switch semanticLevel {
+          case .graphemeCluster:
+            return input[currentPosition].isNewline
+          case .unicodeScalar:
+            return input.unicodeScalars[currentPosition].isNewline
+          }
+        } else {
+          return currentPosition == subjectBounds.upperBound
+        }
+  
+      case .wordBoundary:
```

Comment:
There seem to be no test cases that use the `usesSimpleUnicodeBoundaries` option, should we be testing that?

Reply:
We definitely should be.

@Azoy?

Reply:
We have a couple: `testSimpleWordBoundaries` and https://github.com/apple/swift-experimental-string-processing/blob/96fb215facf425226ca0645d5cab6e84a3befece/Tests/RegexBuilderTests/RegexDSLTests.swift#L540 but the latter was xfail, I have to fix the test because the test is wrong.